### PR TITLE
Add support to defender deploy

### PR DESCRIPTION
**Pull request by MCarlomagno**
## Description

Adds a modal that connects the wizard with the [Defender Deploy Plugin](https://github.com/OpenZeppelin/defender-remix-plugin) (currently only supported in remix).

The modal contains an Iframe that points to the [live version of the remix plugin](https://defender-deploy-wizard.netlify.app/).

### CODE REVIEW

File: packages/ui/src/standalone.css [hunk 1]
```diff
@@ -111,9 +111,9 @@ body {
   margin: -12rem auto 0;
 }
 .wizard {
-  background-color: #f9fafb;
+  /*background-color: #f9fafb;
   box-shadow: 0px 10px 40px rgb(108 133 171 / 20%);
-  border-radius: 1rem;
+   border-radius: 1rem; */
```
[the remark this answers is not in the document]
Comment (ericglau) [reply packages/ui/src/standalone.css:116]: IMO we should remove all of the commented styles throughout this PR if they are no longer relevant.

File: packages/ui/src/App.svelte [hunk 2]
```diff
@@ -313,17 +330,163 @@
...
+      <pre class="flex flex-col grow basis-0 overflow-auto">
+        <code class="hljs grow overflow-auto p-4">{@html highlightedCode}</code>
+      </pre>
+      <DefenderDeployModal isOpen={showDeployModal} />
     </div>
   </div>
 </div>
 
 <style lang="postcss">
+ /* start of the magic */
```
Comment (ericglau) [inline packages/ui/src/App.svelte:390]: Can we put a more detailed comment on what exactly this section does?

File: packages/ui/src/App.svelte [hunk 3]
```diff
@@ -16,23 +16,37 @@
...
     import { injectHyperlinks } from './utils/inject-hyperlinks';
     import { InitialOptions } from './initial-options';
+    import { postMessageToIframe } from './post-message';
+    import { debouncer } from './utils/helpers';
 
     const dispatch = createEventDispatcher();
 
+    // listens to contract changes and posts them to the defender deploy iframe.
+    // we debounce the call to avoid sending too many messages while the user is editing.
+    const debouncedPostMessage = debouncer((contract) => {
```
Comment (ericglau) [inline packages/ui/src/App.svelte:43]: While this is waiting for the debounce timeout, there is no indication to the user that the iframe is waiting to be updated.  The iframe just gets updated 600 ms later, which "feels" like lag.  

@makiopen Any suggestions on how to add some indication that the iframe is pending an update?

**Review by ericglau — COMMENTED** [review]
Looking good so far, thanks!  Just left some initial comments.

I also notice if I close the iframe and re-open it, the API key needs to be entered again.  Do we have a way to persist that information for the current session?

File: packages/ui/src/App.svelte (hunk 2 above)
Comment (MCarlomagno) [reply packages/ui/src/App.svelte:390]: 🧙‍♂️🪄

File: packages/ui/src/App.svelte (hunk 3 above)
Comment (MCarlomagno) [reply packages/ui/src/App.svelte:43]: We can display some loading state with some "compiling contract" message. We would need to make another post message (instantly) to the iframe to activate the loading state until the debounced message arrives

File: packages/ui/src/App.svelte (hunk 2 above)
Comment (makiopen) [reply packages/ui/src/App.svelte]: 🪄

**Comment by MCarlomagno** [comment]
> I also notice if I close the iframe and re-open it, the API key needs to be entered again. Do we have a way to persist that information just for the current session? Or avoid truly closing it, but just hide it.

Thanks! I ended up hiding the iframe instead of conditionally loading the component, a bit hacky but works

**Comment by ericglau** [comment]
The AI Assistant icon also seems missing, not sure where it went.  Perhaps due to the formatting changes.

**Comment by makiopen** [comment]
some z-index issues, brought it back

**Review by ericglau — APPROVED** [review]
Added title (hover) text for "Copy to Clipboard" and "Open in Remix" since they are now icon-only and it may not be obvious what they do.

LGTM, thanks!

File: packages/ui/src/standalone.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (makiopen) [reply packages/ui/src/standalone.css]: removed

File: packages/ui/src/App.svelte (hunk 3 above)
Comment (makiopen) [reply packages/ui/src/App.svelte]: even a spinner would be enough tbh

**Pull request merged**